example3/readme.md

```diff
@@ -0,0 +1,580 @@
+# Summary
+
+An experimental project to implement an OpenTelemetry Metrics API and SDK.
+
+Our goals are:
+
+- To discover/design a simpler approach to Metrics API
+- To explore different SDK implementation based on Metrics API design
+- To experiment with alternative designs/models/ideas for evaluation
+- To complement existing/current .NET runtime design
+- To maintain zero/minimal differences between OpenTelemetry specification and .NET runtime
+- To use this project as an informed starting base for discussion with wider OpenTelemetry community
+
+## Principles
+
+- .NET provides a way to pass datapoints (un-opinionated) in a Time Series.  With
+optional metadata / hints for downstream interpretation.
+
+- .NET may implement and deliver counters (derived from .NET MetricBase) with it's
+own Semantic. (i.e. Counter with Add(), ElapsedDuration as a IDisposable, etc...)
+
+- OTel may implement and deliver counters (derived from .NET MetricBase) with it's
+own Semantic. (i.e. UpCounter with Inc(), ValueRecorder with Record(), etc...)
+
+[Noah]: This principle worries me. I want .NET developers to treat .NET metrics as being 
+a single unified concept. Ie if I am writing a .NET library I want to intrument it:
+a) Without having any reference to an OT specific library
+b) Without mentally apportioning some metrics as being intended for OT usage and others
+are not intended for OT usage.
+
+I'm not opposed to some notion of extensibility for niche use-cases but I would aim
+for 95+% of usage shouldn't need to use it and hopefully it plays no significant role
+in the API design.
+
+- MetricBase will report datapoints into the .NET MetricSource interface
+(EventSource/EventListener model).  OTel SDK will attached as a MetricListeners.
+
+[Noah]: I agree with the Event sourcing principle but I would propose MetricBase is
+the implementation of the source concept in this pattern. So the data flow is:
+MetricBase -> Listener instead of MetricBase -> Source -> Listener.

```

Comment:
Current code has MetricBase -> Listener which implements the EventSource/EventListener pattern.  This WHOLE pair is considered a "source" (maybe I should renamed this to be Provider) which the OTel sdk attaches to for the pipeline building.